sdks/python/apache_beam/runners/portability/portable_runner.py

```diff
@@ -433,13 +435,12 @@ def run_pipeline(self, pipeline, options):
         state_stream,
         cleanup_callbacks)
     if cleanup_callbacks:
-      # We wait here to ensure that we run the cleanup callbacks.
+      # Register an exit handler to ensure cleanup on exit.
+      atexit.register(functools.partial(result._cleanup, on_exit=True))
       _LOGGER.info(
-          'Waiting until the pipeline has finished because the '
-          'environment "%s" has started a component necessary for the '
-          'execution.',
+          'Environment "%s" has started a component necessary for the '
+          'execution. Be sure to call wait_until_finish()',
```

Comment:
We prefer the use of Python's `with` syntax instead of calling `wait_until_finish` explicitly.